examples/dynamic_layer_exchange_example/client.py

```diff
@@ -0,0 +1,75 @@
+from fl4health.utils.load_data import load_cifar10_data
+from fl4health.utils.metrics import Accuracy
+from fl4health.utils.sampler import DirichletLabelBasedSampler
+
+
+class CifarDynamicLayerClient(PartialWeightExchangeClient):
+    def get_data_loaders(self, config: Config) -> Tuple[DataLoader, DataLoader]:
+        batch_size = self.narrow_config_type(config, "batch_size", int)
+        sample_percentage = self.narrow_config_type(config, "sample_percentage", float)
+        beta = self.narrow_config_type(config, "beta", float)
+        assert beta > 0 and 0 < sample_percentage < 1
+        sampler = DirichletLabelBasedSampler(list(range(10)), sample_percentage=sample_percentage, beta=beta)
+        train_loader, val_loader, _ = load_cifar10_data(self.data_path, batch_size, sampler=sampler)
+        return train_loader, val_loader
+
+    def get_criterion(self, config: Config) -> _Loss:
+        return torch.nn.CrossEntropyLoss()
+
+    def get_optimizer(self, config: Config) -> Optimizer:
+        return torch.optim.AdamW(self.model.parameters(), lr=0.001)
+
+    def get_model(self, config: Config) -> nn.Module:
+        return Net().to(self.device)
+
+    def get_parameter_exchanger(self, config: Config) -> ParameterExchanger:
+        norm_threshold = self.narrow_config_type(config, "norm_threshold", float)
+        exchange_percentage = self.narrow_config_type(config, "exchange_percentage", float)
+        normalize = self.narrow_config_type(config, "normalize", bool)
+        select_drift_more = self.narrow_config_type(config, "select_drift_more", bool)
+        filter_by_percentage = self.narrow_config_type(config, "filter_by_percentage", bool)
+
+        layer_selection_function_constructor = LayerSelectionFunctionConstructor(
+            norm_threshold, exchange_percentage, normalize, select_drift_more
+        )
+
+        layer_selection_function = (
+            layer_selection_function_constructor.select_by_percentage()
+            if filter_by_percentage
+            else layer_selection_function_constructor.select_by_threshold()
+        )
+
+        # The user may call layer_selection_function_constructor.select_by_threshold()
+        # to select layers purely based on thresholding instead.
+        parameter_exchanger = DynamicLayerExchanger(layer_selection_function=layer_selection_function)
+
+        return parameter_exchanger
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="FL Client Main")
+    parser.add_argument("--dataset_path", action="store", type=str, help="Path to the local dataset")
+    args = parser.parse_args()
+
+    DEVICE = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+    data_path = Path(args.dataset_path)
+    client = CifarDynamicLayerClient(data_path, [Accuracy("accuracy")], DEVICE, store_initial_model=True)
+    fl.client.start_client(server_address="0.0.0.0:8080", client=client.to_client())
```

Comment:
I got a warning flower that `start_numpy_client` is deprecated and will be removed in the future, and using `start_client` while ensuring `client.to_client()` is called whenever `client` is an instance of `NumpyClient` is advised. We should probably replace all our current calls of `start_numpy_client` in this way.

Reply:
Sounds good. I'll put a ticket in the backlog to do this.